Find a functionally equivalent code:android.bluetooth.BluetoothGatt.writeDescriptor:COMMENT
Method Modifier: public      
Comment:/**
 * Write the value of a given descriptor to the associated remote device.
 *
 * <p>A {@link BluetoothGattCallback#onDescriptorWrite} callback is
 * triggered to report the result of the write operation.
 *
 * <p>Requires {@link android.Manifest.permission#BLUETOOTH} permission.
 *
 * @param descriptor Descriptor to write to the associated remote device
 * @return true, if the write operation was initiated successfully
 */

@@ -1,5 +1,5 @@
 {
-    if (DBG)
+    if (VDBG)
         Log.d(TAG, "writeDescriptor() - uuid: " + descriptor.getUuid());
     if (mService == null || mClientIf == 0)
         return false;
@@ -12,21 +12,27 @@
     BluetoothDevice device = service.getDevice();
     if (device == null)
         return false;
+    synchronized (mDeviceBusy) {
+        if (mDeviceBusy)
+            return false;
+        mDeviceBusy = true;
+    }
     try {
         mService.writeDescriptor(mClientIf, device.getAddress(), service.getType(), service.getInstanceId(), new ParcelUuid(service.getUuid()), characteristic.getInstanceId(), new ParcelUuid(characteristic.getUuid()), descriptor.getInstanceId(), new ParcelUuid(descriptor.getUuid()), characteristic.getWriteType(), AUTHENTICATION_NONE, descriptor.getValue());
     } catch (RemoteException e) {
         Log.e(TAG, "", e);
+        mDeviceBusy = false;
         return false;
     }
     return true;

